includes/models/model.llms.lesson.php

```diff
@@ -364,6 +364,33 @@ public function get_video() {
 		return $this->get_embed( 'video' );
 	}
 
+	/**
+	 * Get the lesson prerequisite
+	 *
+	 * @return int [ID of the prerequisite post]
+	 */
+	public function get_prerequisite() {
+
+		$pre_req = false;
+
+		$course = $this->get_course();
+
+		// Check course level setting
+		if ( $course && $course->must_complete_sequentially() ){
+			// might be false so we are okay
+			$pre_req = $this->get_previous_lesson();
+		}
+
+		// Good ol' fashioned way
+		if ( ! $pre_req ){
+			if ( $this->has_prerequisite ) {
```

Comment:
this could be a pre-existing bug that's gone unnoticed / reported: this line should be checking the method `has_prerequisite()` instead of the property `has_prerequisite()`

Same for the `prerequisite` on the next line.

Reply:
Since we are in the `get_prerequiste()` here I don't think it should use the `has_prerequisite()` method. 

The `has_prerequiste()` method will return true if the course level setting is available OR the lesson has a manually set prerequisite from the course builder.

This means that if we were to use `has_prerequiste()` here we do not know if it's the sequential completion setting or course builder setting.

Also, can't use `get_prerequiste()` that would put us into a recursion.

Does that make sense?

Reply:
Resolved based on your other comment.